app/models/like.rb

```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  # A method that updates the likes counter for a post.
+  def update_likes_counter
+    post = Post.find(post_id)
+    post.update(likes_counter: Like.where(post_id:).count)
+  end
```

Comment:
- [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_likes_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/likes.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

Reply:
Hi @rloterh! One doubt in this one. As the method is supposed to update the likes counter for a post, wouldn't it be written this way since the `likes_counter` is in the Post class?
```
def update_likes_counter
    post.increment!(:likes_counter)
  end
```